### ISSUE TITLE
--recipe flag doesn't work

### ISSUE THREAD

**Issue by faustinoaq** [amberframework/amber#899]
### Description

Ref: https://github.com/amberframework/recipes/issues/32

### Steps to Reproduce

1. `amber new app -r default`

**Expected behavior:** Create an app based on default recipe (or any recipe)

**Actual behavior:** Get an error output:

```
➜  ~ amber new blog -r default
12:11:40 Generate   | (INFO) Rendering App blog in ./blog from default
12:11:41 Generate   | (ERROR) Could not find the recipe default : 302 Found
Environment file not found for ./config/environments/production
```

**Reproduces how often:** 100%

### Versions

Amber v0.8.0

### Additional Information

Looks like this is issue is due crystal `HTTP::Client` doesn't support redirects yet, see: https://github.com/crystal-lang/crystal/issues/2721

/cc @damianham

**Comment by damianham** [comment]
I think we need to move to shards based install asap.

**Comment by faustinoaq** [comment]
> I think we need to move to shards based install asap.

@damianham Yeah, please! `amberframework/recipes` is already a nightmare to maintain :sweat: Should we archive it?

I'm working in a Vue.js recipe, see: https://github.com/faustinoaq/vue-amber-recipe (not published yet)

I think is better to use recipes like this:

```
amber new myapp --recipe https://codeload.github.com/faustinoaq/vue-amber-recipe/zip/0.0.1
```

Just like heroku buildpacks, WDYT?

**Comment by acoustep** [comment]
I had a quick look into fixing this (even if only temporarily while you move to shards based) and it seems that in line 43 of `recipe_fetcher.cr` there is a HTTP request that returns a 302 status. Anything besides a 200 is returning the "Could not find the recipe" response.

Changing this to allow 302 is not enough as it also seems that the HTTP::Client does not follow redirects at the moment (https://github.com/crystal-lang/crystal/issues/2721)

**Comment by acoustep** [comment]
After looking into this it seems that it could be a while before it’s fixed in the Crystal library. Is it worth putting a work-around in for now. E.g. if a 302 response is found send a second request to the redirect location?

I realise this isn’t an efficient solution, but since this command is only ran during development (and only once per project at that) I don’t feel it would be too much of an issue.

If this is an ok solution I can create a fix for this and do a pull request 🙂

**Comment by robacarp** [comment]
@acoustep I think that's whats going to end up in stdlib anyway, at some point, so that sounds good to me.

**Comment by drujensen** [comment]
I think we can just re-open this PR and merge.  https://github.com/amberframework/amber/pull/900

**Comment by damianham** [comment]
I am now out of contract so I will have some time to create a PR to install from shards

**Comment by drujensen** [comment]
@acoustep I merged #900 to master.  Let me know if this fixes the issue.

**Comment by isev-general** [comment]
@drujensen Can confirm it's working 🙂
